### PR TITLE
PR: Implement support for "Barten (1999)" contrast sensitivity function.

### LINKED ISSUE #246: Implement support for "Barten (1999)" contrast sensitivity function.
## References
- http://proceedings.spiedigitallibrary.org/proceeding.aspx?articleid=837000
- http://car.france3.mars.free.fr/Formation%20INA%20HD/HDTV/HDTV%20%202007%20v35/SMPTE%20normes%20et%20confs/Contrastm.pdf
- http://www.umich.edu/~ners580/ners-bioe_481/lectures/pdfs/Barten_ch3s1-9.pdf
- https://www.smpte.org/sites/default/files/2014-05-06-EOTF-Miller-1-2-handout.pdf
- https://www.smpte.org/sites/default/files/23-1615-TS7-2-IProc02-Miller.pdf
- http://www.avsforum.com/forum/139-display-calibration/2020689-need-help-barten-contrast-sensitivity-function.html

### PR DESCRIPTION
**Pull request by KelSolaar**
Closes #246.

### CODE REVIEW

**Comment by KelSolaar** [comment]
Here is a test with a maximised spatial frequency *u* versus fixed *u = 15*:

image

It is actually going way below 2% for low luminance values, fixed *u* was closer to *Figure 31*. I'm starting to wonder what settings were used to generate the figure.

Here is the code used to generate mine:

```python
# %%
from __future__ import division

import numpy as np
from scipy.optimize import fmin

from colour.plotting import colour_style, plot_multi_functions
from colour.contrast.barten1999 import (
    function_contrast_sensitivity_Barten1999, pupil_diameter_Barten1999,
    retinal_illuminance_Barten1999, sigma_Barten1999)

_ = colour_style()

# %%
settings_BT2246 = {
    'k': 3.0,
    'T': 0.1,
    'X_max': 12,
    'N_max': 15,
    'n': 0.03,
    'p': 1.2274 * 10**6,
    'phi_0': 3 * 10**-8,
    'u_0': 7,
}


def maximise_spatial_frequency(L):
    maximised_spatial_frequency = []
    for L_v in L:
        X_0 = 60
        d = pupil_diameter_Barten1999(L_v, X_0)
        sigma = sigma_Barten1999(0.5 / 60, 0.08 / 60, d)
        E = retinal_illuminance_Barten1999(L_v, d, True)
        CSF = lambda x: -function_contrast_sensitivity_Barten1999(
            u=x,
            sigma=sigma,
            X_0=X_0,
            E=E,
            **settings_BT2246)

        maximised_spatial_frequency.append(fmin(CSF, 0, disp=False)[0])

    return maximised_spatial_frequency


L = np.logspace(np.log10(0.01), np.log10(100), 100)

X_0 = 60
d = pupil_diameter_Barten1999(L, X_0)
sigma = sigma_Barten1999(0.5 / 60, 0.08 / 60, d)
E = retinal_illuminance_Barten1999(L, d, True)
u = maximise_spatial_frequency(L)

y1 = lambda x: 1 / function_contrast_sensitivity_Barten1999(
    u=u,
    sigma=sigma,
    X_0=X_0,
    E=E,
    **settings_BT2246) * 2 * (1/ 1.27)

y2 = lambda x: 1 / function_contrast_sensitivity_Barten1999(
    u=X_0 / 15,
    sigma=sigma,
    X_0=X_0,
    E=E,
    **settings_BT2246) * 2 * (1/ 1.27)


plot_multi_functions(
    {'Maximised U': y1, 'Fixed U': y2},
    samples=L,
    log_x=10,
    # log_y=10,
    bounding_box=[0.1, 100, 0.001, 0.025],
    **{'axes.grid.which': 'both'})

```

**Comment by KelSolaar** [comment]
It works! I was actually clipping the X-axis:

image

**Comment by KelSolaar** [comment]
@nick-shaw, @MichaelMauderer Any chances to look at that? Pretty sure you started @nick-shaw, should we merge?

**Comment by KelSolaar** [comment]
Fixed the _BIBLIOGRAPHY.bib_ conflict by rebasing on top of `develop`, it is not super elegant but at this stage and given this file acts almost like a binary file where changes can happen anywhere without much control, it is the path of least resistance.

File: colour/contrast/__init__.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Spatial frequency :math:`u`, the cycles per degree.
+    u_0 : numeric or array_like, optional
+        {:func:`colour.contrast.contrast_sensitivity_function_Barten1999`},
+        Spatial frequency :math:`u_0` in :math:`cycles\\div degrees` above
+        which the lateral inhibition ceases.
+
+    Returns
+    -------
+    ndarray
+        Contrast sensitivity function :math:`S`.
```
Comment (MichaelMauderer) [inline colour/contrast/__init__.py]: I did not read the paper, so I'm wondering why a "function" has the type ndarray?
Might be good to elaborate a bit more what the array represents. The documentaiton of the function doesn't seem to explain it much more.

Comment (KelSolaar) [reply colour/contrast/__init__.py]: Ah, nicely spotted, should read _Contrast sensitivity :math:`S`._

**Comment by MichaelMauderer** [comment]
Is there a unit associated with it? (cycles per degree or something like
that?)

**Comment by KelSolaar** [comment]
I don't think there is, Barten (1999) says it is a dimensionless quantity:

image

**Comment by KelSolaar** [comment]
I just force-pushed to integrate the copyright date changes and updated the docstrings.

**Comment by KelSolaar** [comment]
Thanks @MichaelMauderer!

**Pull request merged**